public/app/plugins/datasource/grafana/datasource.ts

```diff
@@ -135,14 +134,14 @@ export class GrafanaDatasource extends DataSourceWithBackend<GrafanaQuery> {
           buffer.maxDelta = request.range.to.valueOf() - request.range.from.valueOf();
         }
 
-        results.push(
-          getGrafanaLiveSrv().getDataStream({
-            key: `${request.requestId}.${counter++}`,
-            addr: addr!,
-            filter,
-            buffer,
-          })
-        );
+        const opts = {
+          key: `live-${target.refId}-`,
+          addr: addr!,
+          filter,
+          buffer,
+        };
+        opts.key += djb2Hash(JSON.stringify(opts)); // Keep a constant key for the same query+options
```

Comment:
can we cache the opts string instead of doing `JSON.stringify(opts)` each time?

Reply:
it is a nested structure, so i guess we can walk it iterating keys+values?  but not sure that is any better